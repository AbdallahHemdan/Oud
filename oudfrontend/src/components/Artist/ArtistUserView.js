import React, { Component } from "react";
import UpperContainer from "./UpperContainer/UpperContainer";
import LowerContainer from "./LowerContainer/LowerContainer";
import { base } from "../../config/environment";
import { getRequest, deleteRequest, putRequest } from "../../utils/requester";
import PropTypes from "prop-types";
import AddToPlaylist from "../commonComponents/addToPlaylist/addToPlaylist";
/**
 * A class component to control rendering the upper and lower parts of the artist page.
 * @author Ahmed Ashraf
 * @component
 * @example
 * return (
 *  <ArtistUserView artistId={"1"}/>
 * )
 */
class ArtistUserView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      id: "",
      img: "",
      photo: "",
      signInId: "",
      followStatus: false,
      bio: "",
      displayAdd: false
    };
  }
  /**
   * After the component mount, fetch the needed artist data
   * @func
   * @returns {void}
   */
  componentDidMount() {
    console.log(`${base}/artists/${this.props.artistId}`);
    getRequest(`${base}/artists/${this.props.artistId}`)
      .then(response => {
        this.setState({
          id: response.data.id,
          username: response.data.displayName,
          bio: response.data.bio,
          img: response.data.images[1]
        });
        getRequest(
          `${base}/me/following/containes?type=artist&ids=[${this.props.artistId}]`
        )
          .then(response => {
            this.setState({ followStatus: response.data.ids[0] });
          })
          .catch(error => {
            console.log(error);
          });
      })
      .catch(error => {
        console.log(error.response);
      });

    getRequest(`${base}/me`)
      .then(response => {
        this.setState({ signInId: response.data.id });
      })
      .catch(error => {
        console.log(error);
      });
  }
  /**
   * If the component updated, update the artist id and username
   * @func
   * @returns {void}
   */
  componentDidUpdate(prevProps, prevState) {
    if (prevProps.artistId !== this.props.artistId) {
      getRequest(`${base}/artists/${this.props.artistId}`)
        .then(response => {
          this.setState({
            id: response.data.id,
            username: response.data.displayName
          });
        })
        .catch(error => {
          console.log(error);
        });
    }
  }
  /**
   * A function to handle the follow actions. It toggle the button and make a request to the server
   * @func
   * @returns {void}
   */
  handleFollowClick = event => {
    /*
     1) make put request if it was false 
     2) make delet request if true
     3) then change the state
    */

    if (this.state.followStatus) {
      /*this shouild be in route me/following/ids=*,*,*,*&type=user/artist*/
      deleteRequest(`${base}/me/following?type=artist&ids[${this.state.id}]`)
        .then(response => {
          console.log(response);
        })
        .catch(error => console.log(error));
    } else {
      putRequest(`${base}/me/following?type=artist`, {
        ids: [this.state.id]
      })
        .then(response => {
          console.log(response);
        })
        .catch(error => {
          console.log(error);
        });
    }
    this.setState({ followStatus: !this.state.followStatus });
  };
  /**
   * A utility for the AddToPlaylist component to display the component
   * @func
   * @returns {void}
   */
  addToPlaylist = () => {
    this.setState({ displayAdd: true });
  };
  /**
   * A utility for the AddToPlaylist component to close the component
   * @func
   * @returns {void}
   */
  closeAddToPlaylist = () => {
    this.setState({ displayAdd: false });
  };
  render() {
    return (
      <div
        className="artist-user-view-page"
        data-testid="artist-user-view-page"
      >
        {this.state.displayAdd && (
          <AddToPlaylist
            data-testid="artist-top-tracks"
            display={this.state.displayAdd}
            close={this.closeAddToPlaylist}
          />
        )}
        <div className="profile-user" data-testid="artist-user-view">
          <UpperContainer
            data-testid="artist-upper-container"
            artistId={this.props.artistId}
            handleFollowClick={this.handleFollowClick}
            followStatus={this.state.followStatus}
            username={this.state.username}
            cover={this.state.img}
          />
          <LowerContainer
            data-testid="artist-lower-container"
            artistId={this.props.artistId}
            bio={this.state.bio}
            addToPlaylist={this.addToPlaylist}
            username={this.state.username}
            img={this.state.img}
          />
        </div>
      </div>
    );
  }
}
ArtistUserView.propTypes = {
  /**
   * The unique idetifier of the author
   */
  artistId: PropTypes.string.isRequired
};
export default ArtistUserView;
