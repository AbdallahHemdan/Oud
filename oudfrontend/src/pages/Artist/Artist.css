.play-artist {
  float: left;
  background-color: #f9b835;
  color: white;
  border-radius: 28px;
  font-weight: 700;
  font-size: 14px;
  width: 100px;
  margin: 0;
  padding: 5px;
  align-self: flex-start;
  transition: transform 0.2s;
  align-self: flex-end;
  margin-top: 8px;
  margin-right: 10px;
}

.play-artist:hover {
  transform: scale(1.05);
  background-color: #f9b835;
  color: white;
}
.play-artist:active,
.play-artist:focus {
  background-color: transparent;
  color: #f9b835;
}
.artist-user .upperContainerFollowingButton {
  float: left;
  margin-right: 10px;
}
.artist-user .upperContainerFollowButton {
  float: left;
  margin-right: 10px;
}
.artist-user .upperContainerProfile {
  -webkit-box-align: center;
  position: relative;
  align-items: center;
  background-size: cover;
  background-position: 50%;
}
/* .artist-user .upperContainerProfile::before {
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background: -webkit-gradient(
    linear,
    left top,
    left bottom,
    color-stop(-30%, transparent),
    to(#121212)
  );
  background: linear-gradient(transparent -30%, #121212);
  box-sizing: initial;
} */
.artist-user .upperContainerProfile h1 {
  font-size: 50px;
  font-weight: 1000;
  transition: 0.25s;
}
.artist-ellipsis-container .ellipsis-icon {
  visibility: visible;
  margin-top: 8px;
}
.artist-user .ellipsis-img {
  float: left;
}
.artist-bio {
  color: #a9a1a1;
  width: 70%;
}
.artist-pop {
  /* width: 100%; */
  min-width: 100% !important;
}
.artist-user .upperNav {
  z-index: 1;
}
.artist-user .userName-profile-scrolled {
  margin-left: 30px;
}
.artis-overview .SEE-ALL-Overview {
  cursor: pointer;
}
.artist-appears-on,
.artist-singles,
.artist-compilation {
  margin-top: 2%;
}
/* .artist-options-dropdown .dropdwon {
  float: left;
} */
.artist-user .dropdown-menu {
  background-color: black;
}
.artist-user .dropdown-item {
  color: white;
  cursor: pointer;
}
.artist-user .dropdown-item:hover {
  background-color: #ffc107;
}
.swal2-icon.swal2-success {
  border-color: #ffce00;
  color: #ffce00;
}
.swal2-icon.swal2-success [class^="swal2-success-line"] {
  background-color: #ffce00;
}
.artist-user-view-page .createPlaylist {
  z-index: 4;
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  height: 100%;
}
.artist-user-view-page {
  width: 100%;
}
.artist-user-view-page .profile-user {
  /* width: 87%; */
}
.artis-overview .col-xs-8,
.artis-overview .col-md-6,
.artis-overview .col-lg-8,
.artis-overview .col-xl-8 {
  width: 100%;
  min-width: 100%;
}
