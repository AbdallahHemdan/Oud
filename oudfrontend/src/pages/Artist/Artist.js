import React from "react";
import Sidebar from "../../components/Sidebar/Sidebar";
import Navbar from "../../components/Navbar/Navbar";
import ArtistUserView from "../../components/Artist/ArtistUserView";
import "./Artist.css";
/**
 * A class component to render the artist page
 * @author Ahmed Ashraf
 * @component
 * @example
 * return (
 *  <Artist />
 * )
 */
function Artist(props) {
  console.log(props.match.params.artistId);
  return (
    <div className="dummyParent">
      <Sidebar />
      <Navbar isLoggedIn={true} />
      <ArtistUserView
        data-test="artist-upper-container"
        artistId={props.match.params.artistId}
      />
      {/* <ActivityBar /> */}
    </div>
  );
}
export default Artist;
