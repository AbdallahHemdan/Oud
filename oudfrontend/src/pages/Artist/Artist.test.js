import React from "react";
import { shallow } from "enzyme";
import Artist from "./Artist";
const setup = () => {
  const props = {
    match: {
      params: {
        artistId: "1",
      },
    },
  };
  const component = shallow(<Artist {...props} />);
  return component;
};

describe("artist snapshot test", () => {
  it("renders correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
