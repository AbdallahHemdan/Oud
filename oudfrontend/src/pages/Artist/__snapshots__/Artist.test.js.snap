// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`artist snapshot test renders correctly 1`] = `ShallowWrapper {}`;
